zospy/analyses/raysandspots.py

```diff
@@ -0,0 +1,258 @@
+import os
+import re
+from tempfile import mkstemp
+from io import StringIO
+
+import numpy as np
+import pandas as pd
+
+from zospy import utils
+from zospy.analyses.base import AnalysisResult, AttrDict
+from zospy.api import constants
+
+
+def _structure_ray_trace_data_result(line_list):
+    """Structures the result of a ray trace data report.
+
+    Parameters
+    ----------
+    line_list: list of str
+        The line list obtained by reading in the results
+
+    Returns
+    -------
+    dict
+        The results structured in a dictionary
+    """
+    # Create output dict
+    res = AttrDict()
+
+    # Add header
+    hdr = line_list[0].strip().replace('\ufeff', '')
+    res['Header'] = hdr
+
+    # Register landmarks
+    landmarks = ['Real Ray Trace Data',
+                 'Paraxial Ray Trace Data',
+                 'Trace of Paraxial Y marginal, U marginal, Y chief, U chief only']
+    comp = re.compile(rf'^({"|".join(landmarks)})', re.IGNORECASE)
+    inds = {comp.match(line).group().lower(): {'rnum': rnum, 'name': comp.match(line).group()}
+            for rnum, line in enumerate(line_list) if comp.match(line)}
+
+    order_of_appearance = [item[0] for item in sorted(inds.items(), key=lambda x: x[1]['rnum'])]
+
+    # Add general lens data
+    for target in inds.keys():
+        section_start = inds[target]['rnum'] + 1  # + 1 to ignore section title
+
+        # Get endpoint
+        if order_of_appearance.index(target) == len(inds) - 1:
+            section_end = len(line_list)
+        else:
+            target2 = order_of_appearance[order_of_appearance.index(target) + 1]
+            section_end = inds[target2]['rnum']
+
+        # Read as dataframe

```

Comment:
@andibarg On certain machines, OpticStudio provides it's output with a comma rather than a point as decimal separator. In those cases, the numbers are strings rather than floats in the resulting dataframe. I think this can easily be resolved by adding ```.replace(',', '.')```. However, in some edge cases I ran into problems with this and I implemented some regex to find the comma delimiters (see [reports.py](https://github.com/MREYE-LUMC/ZOSPy/blob/c9490a82f31d247f3b4d2ec192c567680041d03a/zospy/analyses/reports.py)). 

Are you aware of any cases in which the text output would contain commas sthat are not intended to be decimal separators?

Reply:
@LucVV Good point and yes. In the single ray trace text output, there could be commas in the comment column of the different surfaces. The ray fan text may also include comments.

Perhaps instead of the regex solution, we could make use of the decimal option in `pd.read_csv(Input, decimal=',')`. Apparently, the system default decimal can be found with `import locale` and `locale.localeconv()["decimal_point"]`. What do you think?